drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -342,6 +379,612 @@ static int mcux_flexcomm_uart_config_get(const struct device *dev,
+
+	/* Getting dma status to use to calculate bytes sent */
+	struct dma_status status = {0};
+
+	ret = dma_get_status(config->tx_dma.dev, config->tx_dma.channel, &status);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Done with the DMA transfer, can stop it now */
+	ret = dma_stop(config->tx_dma.dev, config->tx_dma.channel);
+	if (ret) {
+		return ret;
+	}
+
+	/* Define TX abort event before resetting driver variables */
+	size_t sent_len = data->tx_data.xfer_len - status.pending_length;
+	const uint8_t *aborted_buf = data->tx_data.xfer_buf;
+	struct uart_event tx_abort_event = {
+		.type = UART_TX_ABORTED,
+		.data.tx.buf = aborted_buf,
+		.data.tx.len = sent_len
+	};
+
+	/* Driver data needs reset since there is no longer an ongoing
+	 * transfer, this should before the user callback, not after,
+	 * just in case the user callback calls tx again
+	 */
+	data->tx_data.xfer_len = 0;
+	data->tx_data.xfer_buf = NULL;
+
+	async_user_callback(dev, &tx_abort_event);
+
+	return ret;
+}
+
+static int mcux_flexcomm_uart_rx_enable(const struct device *dev, uint8_t *buf,
+					const size_t len, const int32_t timeout)
+{
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+	int ret = 0;
+
+	if (config->rx_dma.dev == NULL) {
+		return -ENODEV;
+	}
+
+	/* Getting DMA status to tell if channel is busy or not set up */
+	struct dma_status status;
+
+	ret = dma_get_status(config->rx_dma.dev, config->rx_dma.channel, &status);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* There is an ongoing transfer */
+	if (status.busy) {
+		return -EBUSY;
+	}
+
+	/* Disable RX DMA requests for uart while setting up */
+	USART_EnableRxDMA(config->base, false);
+
+	/* Set up the dma channel/transfer */
+	data->rx_data.xfer_buf = buf;
+	data->rx_data.xfer_len = len;
+	data->rx_data.active_block.dest_address = (uint32_t)data->rx_data.xfer_buf;
+	data->rx_data.active_block.source_address = (uint32_t) &config->base->FIFORD;
+	data->rx_data.active_block.block_size = data->rx_data.xfer_len;
+
+	ret = dma_config(config->rx_dma.dev, config->rx_dma.channel,
+				(struct dma_config *) &config->rx_dma.cfg);
+	if (ret) {
+		return ret;
+	}
+
+	data->rx_data.timeout = timeout;
+
+	/* Enable RX DMA requests from UART */
+	USART_EnableRxDMA(config->base, true);
+
+	/* Enable start bit detected interrupt, this is the only
+	 * way for the flexcomm uart to support the Zephyr Async API.
+	 * This is only needed if using a timeout.
+	 */
+	if (timeout != SYS_FOREVER_US) {
+		config->base->INTENSET |= USART_INTENSET_STARTEN_MASK;
+	}
+
+	/* Trigger the DMA to start transfer */
+	ret = dma_start(config->rx_dma.dev, config->rx_dma.channel);
+	if (ret) {
+		return ret;
+	}
+
+	/* Request next buffer */
+	struct uart_event rx_buf_request = {
+		.type = UART_RX_BUF_REQUEST,
+	};
+
+	async_user_callback(dev, &rx_buf_request);
+
+	return ret;
+}
+
+static void flexcomm_uart_rx_update(const struct device *dev)
```

Comment:
Should we call this `mcux_flexcomm_uart_rx_update` to be consistent with the naming of functions in this file.